gusto/diagnostics.py

```diff
@@ -270,6 +270,42 @@ def compute(self, state):
         return self.field.interpolate(rho * (cv * T + c_vv * w_v * T + c_pv * w_c * T - (L_v0 - (c_pl - c_pv) * (T - T_0)) * w_c))
 
 
+class HydrostaticImbalance(DiagnosticField):
+    name = "HydrostaticImbalance"
+
+    def setup(self, state):
+        if not self._initialised:
+            space = state.spaces("Vv")
+            super(HydrostaticImbalance, self).setup(state, space=space)
+            rho = state.fields("rho")
+            rhobar = state.fields("rhobar")
+            theta = state.fields("theta")
+            thetabar = state.fields("thetabar")
+            pi = exner(theta, rho, state)
+            pibar = exner(thetabar, rhobar, state)
+
+            cp = Constant(state.parameters.cp)
+            n = FacetNormal(state.mesh)
+
+            F = TrialFunction(space)
+            w = TestFunction(space)
+            a = inner(w, F)*dx
+            L = (- cp*div((theta-thetabar)*w)*pibar*dx
+                 + cp*jump((theta-thetabar)*w, n)*avg(pibar)*dS_v
+                 - cp*div(thetabar*w)*(pi-pibar)*dx
+                 + cp*jump(thetabar*w, n)*avg(pi-pibar)*dS_v)
+
+            bcs = [DirichletBC(space, 0.0, "bottom"),
+                   DirichletBC(space, 0.0, "top")]
+
+            imbalanceproblem = LinearVariationalProblem(a, L, self.field, bcs=bcs)
+            self.imbalance_solver = LinearVariationalSolver(imbalanceproblem)
+
+    def compute(self, state):
+        self.imbalance_solver.solve()
+        return self.field[1]
```

Comment:
The y-component of the imbalance?

Reply:
Note this is almost certainly not a `Function`, but rather a ufl expression?

Reply:
The imbalance lives in the space of vector fields with zero horizontal component. The way it is written now only works in a 2D setting, so I should probably write something like ```self.field[-1]``` or ```self.field[dimension_of_problem -1]```

```self.field``` is the function I am writing my solution of the variational problem into (like in the ```Vorticity``` diagnostic), and I thought ```self.field[1]``` would then simply give me the vertical component of this function (like in the ```VelocityZ``` diagnostic)

This works, except of the output is a discontinuous field, but I thought we are more interested in the values than the field's shape (like in the ```CourantNumber``` diagnostic), so I left it like this.